### PR TITLE
rustc_error, rustc_private: Switch to stable hash containers

### PR DESCRIPTION
**Pull request by NiklasJonsson**
Relates https://github.com/rust-lang/rust/issues/84447

### CODE REVIEW

File: compiler/rustc_privacy/src/lib.rs [hunk 1]
```diff
@@ -2080,7 +2079,7 @@ fn check_private_in_public(tcx: TyCtxt<'_>, (): ()) {
     };
     tcx.hir().walk_toplevel_module(&mut visitor);
 
-    let mut old_error_set_ancestry = HirIdSet::default();
+    let mut old_error_set_ancestry = FxIndexSet::<HirId>::default();
```
Comment (compiler-errors) [inline compiler/rustc_privacy/src/lib.rs:2082]: I wonder if we should just change `HirIdSet` to be a `FxIndexSet<HirId>`

Comment (NiklasJonsson) [reply compiler/rustc_privacy/src/lib.rs:2082]: Yeah, I considered it but I wasn't sure of the effects on other code. I'll give it a try!

**Comment by NiklasJonsson** [comment]
> Looks like inlining noise?

What does this mean? That the heuristics governing inlining are triggering differently and creating noise in the benchmark?

**Comment by pnkfelix** [comment]
discussed in today's T-compiler triage meeting.

That regression (to a single secondary benchmark `deeply-nested-multi`) should not block moving this work forward.

I skimmed over it and it looks okay to me.

So r=me once conflicts are resolved.

**Comment by oli-obk** [comment]
yea, I have no idea what else it could be from that cachegrind run. I'd say we take the "hit". It may randomly unregress with other changes again.

**Comment by oli-obk** [comment]
Sorry for forgetting about this PR. Please rebase and we'll send it off

@bors rollup=never

**Comment by NiklasJonsson** [comment]
No worries, I haven't been very active for a few months anyway.

@rustbot ready

**Pull request merged**